packages/core/versioning/core-version-http-server/src/example.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { schema } from '@kbn/config-schema';
+import type { IRouter, RequestHandlerContextBase } from '@kbn/core-http-server';
+import type { VersionHTTPToolkit } from './version_http_toolkit';
+
+interface MyCustomContext extends RequestHandlerContextBase {
+  fooService: { create: (value: string, id: undefined | string, name?: string) => Promise<void> };
+}
+const vtk = {} as unknown as VersionHTTPToolkit;
+const router = {} as unknown as IRouter<MyCustomContext>;
+
+const versionedRouter = vtk.createVersionedRouter({ router });
+
+// @ts-ignore unused variable
+const versionedRoute = versionedRouter
+  .post({
+    path: '/api/my-app/foo/{fooId?}/{name?}', // <= design mistake, {name?} should be a query parameter, but it has been released...
+    options: { timeout: { payload: 60000 } },
+  })
+  .addVersion(
+    {
+      version: '1',
+      validate: {
+        params: schema.object({
+          fooId: schema.maybe(schema.string({ minLength: 10, maxLength: 13 })),
+          name: schema.maybe(schema.string({ minLength: 2, maxLength: 50 })),
+        }),
+        body: schema.object({ foo: schema.string() }),
+      },
+    },
+    async (ctx, req, res) => {
+      await ctx.fooService.create(req.body.foo, req.params.fooId, req.params.name);
+      return res.ok({ body: { foo: req.body.foo } });
+    }
+  )
+  // BREAKING CHANGE: { foo: string } => { fooString: string } in body
+  .addVersion(
+    {
+      version: '2',
+      path: '/api/my-app/foo/{id?}/{name?}', // Update "fooId" => "id", this is not a breaking change!
```

Comment:
Why is it not a breaking change? Our code won't be able to access `params.fooId`, the value will now be under `params.id` right?

Reply:
breaking changes refer to changes with break clients, we can always adapt our code to deal with "internal" breaking changes like this. So a client will still construct exactly the same query whether we map that path variable to `id` or `fooId`